EXAMPLES.md

```diff
@@ -0,0 +1,330 @@
+    })
+    .WithAccessToken(options =>
+    {
+        options.Audience = Configuration["Auth0:Audience"];
+    });
+```
+
+Apart from being able to configure the audience globally, the SDK's `LoginAuthenticationPropertiesBuilder` can be used to supply the audience when triggering login through `HttpContext.ChallengeAsync`:
+
+```csharp
+var authenticationProperties = new LoginAuthenticationPropertiesBuilder()
+    .WithRedirectUri("/") // "/" is the default value used for RedirectUri, so this can be omitted.
+    .WithAudience("YOUR_AUDIENCE")
+    .Build();
+
+await HttpContext.ChallengeAsync(Auth0Constants.AuthenticationScheme, authenticationProperties);
+```
+
+> :information_source: Specifying the Audience when calling `HttpContext.ChallengeAsync` will take precedence over any globally configured Audience.
+
+### Retrieving the Access Token
+
+As the SDK uses the OpenId Connect middleware, the ID Token is decoded and the corresponding claims are added to the `ClaimsIdentity`, making them available by using `User.Claims`.
+
+The access token can be retrieved by calling `HttpContext.GetTokenAsync("access_token")`.
+
+```csharp
+[Authorize]
+public async Task<IActionResult> Profile()
+{
+    var accessToken = await HttpContext.GetTokenAsync("access_token");
+
+    return View(new UserProfileViewModel()
+    {
+        Name = User.Identity.Name,
+        EmailAddress = User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email)?.Value,
+        ProfileImage = User.Claims.FirstOrDefault(c => c.Type == "picture")?.Value
+    });
+}
+```
+
+### Refresh Tokens
+
+In the case where the application needs to use an Access Token to access an API, there may be a situation where the Access Token expires before the application's session does. In order to ensure you have a valid Access Token at all times, you can configure the SDK to use Refresh Tokens:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddAuth0WebAppAuthentication(options =>
+        {
+            options.Domain = Configuration["Auth0:Domain"];
+            options.ClientId = Configuration["Auth0:ClientId"];
+            options.ClientSecret = Configuration["Auth0:ClientSecret"];
+        })
+        .WithAccessToken(options =>
+        {
+            options.Audience = Configuration["Auth0:Audience"];
+            options.UseRefreshTokens = true;
+        });
+}
+```
+
+#### Detecting the absense of a Refresh Token
+
+In the event where the API, defined in your Auth0 dashboard, isn't configured to [allow offline access](https://auth0.com/docs/get-started/dashboard/api-settings), or the user was already logged in before the use of Refresh Tokens was enabled (e.g. a user logs in a few minutes before the use of refresh tokens is deployed), it might be useful to detect the absense of a Refresh Token in order to react accordingly (e.g. log the user out locally and force them to re-login).
```

Comment:
```suggestion
In the event where the API, defined in your Auth0 dashboard, isn't configured to [allow offline access](https://auth0.com/docs/get-started/dashboard/api-settings), or the user was already logged in before the use of refresh tokens was enabled (e.g. a user logs in a few minutes before the use of refresh tokens is deployed), it might be useful to detect the absense of a refresh token in order to react accordingly (e.g. log the user out locally and force them to re-login).
```